java/core/src/main/java/co/worklytics/psoxy/gateway/HttpEventRequest.java

```diff
@@ -14,6 +14,8 @@
 
     Optional<String> getQuery();
 
-    Optional<List<String>> getHeader(String headerName);
+    Optional<String> getHeader(String headerName);
+
+    Optional<List<String>> getMultiValueHeader(String headerName);
```

Comment:
I don't think it's logical to provide *two* access points to the same thing.  a single-value header is just a special case of a multi-valued one. Callers should always us the `multi-value` case, to ensure they're handling it properly - rather than calling `getHeader()` and having ambiguous behavior in multi-valued case

so imho, `Optional<List<String>> getHeader(String headerName)` alone

Reply:
:smile: not gonna agree. 99.9% of cases single values headers, adding the burden of picking first to the client doesn't make any sense. Call it syntactic sugar if you want, but simplifies code and the caller must know if it's dealing with a multi-valued header or not explicitly.